compiler/rustc_ast/src/token.rs

```diff
@@ -553,6 +553,10 @@ impl Token {
         self == &Lt || self == &BinOp(Shl)
     }
 
+    pub fn is_inline_const_start(&self) -> bool {
+        self.is_keyword(kw::Const)
+    }
```

Comment:
Could you always use `check_inline_const` instead of introducing new functions?
You could add a `dist: usize` to it for the lookahead case (see `fn is_keyword_ahead` as an example.)